<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <ros2_control name="ArduinoHardware" type="system">
    <hardware>
      <plugin>arm_hardware/ArmHardwareInterface</plugin>
      <param name="serial_port">/dev/ttyACM0</param>
    </hardware>

    <joint name="waist_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="shoulder_pitch_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="elbow_pitch_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="wrist_pitch_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="gripper_finger_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
  </ros2_control>
</robot>